### ISSUE TITLE
Booleans in expressions lead to error

### ISSUE THREAD

**Issue by MichaelClerx** [ModellingWebLab/cellmlmanip#350]
```
<?xml version="1.0" encoding="UTF-8"?>
<!-- CellML Test Suite. https://github.com/MichaelClerx/cellml-validation -->
<!-- CellML 1.0, 4.2.3: The MathML "CellML subset"
Logical constants: true and false -->
<model name="mathml_logic_constants"
       xmlns="http://www.cellml.org/cellml/1.0#"
       xmlns:cellml="http://www.cellml.org/cellml/1.0#">
  <component name="A">
    <variable name="x" units="dimensionless" initial_value="0" />
    <variable name="iftrue" units="dimensionless" />
    <variable name="iffalse" units="dimensionless" />
    <math xmlns="http://www.w3.org/1998/Math/MathML">
      <apply>
        <eq/>
        <ci>iftrue</ci>
        <piecewise>
          <piece>
            <cn cellml:units="dimensionless">1</cn>
            <apply>
              <eq />
              <apply>
                <eq />
                <ci>x</ci>
                <cn cellml:units="dimensionless">0</cn>
              </apply>
              <true />
            </apply>
          </piece>
          <otherwise>
            <cn cellml:units="dimensionless">0</cn>
          </otherwise>
        </piecewise>
      </apply>
      <apply>
        <eq/>
        <ci>iffalse</ci>
        <piecewise>
          <piece>
            <cn cellml:units="dimensionless">1</cn>
            <apply>
              <eq />
              <apply>
                <eq />
                <ci>x</ci>
                <cn cellml:units="dimensionless">0</cn>
              </apply>
              <false />
            </apply>
          </piece>
          <otherwise>
            <cn cellml:units="dimensionless">0</cn>
          </otherwise>
        </piecewise>
      </apply>
    </math>
  </component>
</model>
```

results in

```
'BooleanTrue' object has no attribute 'as_coeff_Add'
```

**Comment by MauriceHendrix** [comment]
looks like a sympy bug

```
>>> eqs
(_1, Eq(Eq(_Ax, _0), True))
>>> sympy.Piecewise(eqs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\functions\elementary\piecewise.py", line 145, in __new__
    r = cls.eval(*newargs)
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\functions\elementary\piecewise.py", line 190, in eval
    cond = cond.replace(
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\core\basic.py", line 1577, in replace
    rv = walk(self, rec_replace)
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\core\basic.py", line 1562, in walk
    rv = F(rv)
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\core\basic.py", line 1570, in rec_replace
    v = _value(expr, result)
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\core\basic.py", line 1536, in <lambda>
    _value = lambda expr, result: value(expr)
  File "C:\Users\uczmh2\Desktop\chaste_codegen\lib\site-packages\sympy\core\relational.py", line 52, in _canonical_coeff
    b, l = rel.lhs.as_coeff_Add(rational=True)
AttributeError: 'BooleanTrue' object has no attribute 'as_coeff_Add'
```

**Comment by MauriceHendrix** [comment]
**This issue seems to be happening with 1.10.1, but does not happenw ith sympy 1.9**
```
(chaste_codegen) C:\Users\uczmh2\Desktop>pip install sympy==1.9
....
>>> import cellmlmanip as cm
>>> cm.load_model('C:\\Users\\uczmh2\\Desktop\\test.cellml')
[(_1, Eq(Eq(_A$x, _0), True)), (_0, True)]
[(_1, Eq(Eq(_A$x, _0), False)), (_0, True)]
<cellmlmanip.model.Model object at 0x0000021E3AACFF10>
```

**Comment by MichaelClerx** [comment]
Above error was with
```
(venv) [michael@fedora validation]$ python
Python 3.10.7 (main, Sep  7 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sympy
>>> sympy.__version__
'1.11.1'
```

**Comment by MichaelClerx** [comment]
Doesn't happen with sympy 1.9

**Comment by MauriceHendrix** [comment]
fixed by #353
